core/log/log.odin

```diff
@@ -9,26 +12,75 @@ Level :: enum {
 }
 
 Option :: enum {
-	Level,
-	Time,
-	File,
-	Line,
-	Procedure,
+    Level,
+    Date,
+    Time,
+    Short_File_Path,
+    Long_File_Path,
+    Line,
+    Procedure,
+    Terminal_Color
 }
+
 Options :: bit_set[Option];
+Full_Timestamp_Opts :: Options{
+    Option.Date,
+    Option.Time
+};
+Location_Header_Opts :: Options{
+    Option.Short_File_Path,
+    Option.Long_File_Path,
+    Option.Line,
+    Option.Procedure,
+};
+Location_File_Opts :: Options{
+    Option.Short_File_Path,
+    Option.Long_File_Path
+};
 
 Logger_Proc :: #type proc(data: rawptr, level: Level, ident, text: string, options: Options, location := #caller_location);
 
 Logger :: struct {
 	procedure: Logger_Proc,
 	data:      rawptr,
+    options:   Options,
```

Comment:
Please try to not add the extra data here, keep that data in the `data` component if possible. If it isn't, then fine but I do want to keep this "interface data" small.

Reply:
Options and ident will have to be here if they're going to be passed to the logger proc.

Reply:
🤔 I wonder if they should be passed or let the logger decide that itself.

Reply:
It probably is a good idea my question is whether `options` and `ident` will be modified enough that they should be there or that we always want those options on all loggers.